src/storage/storage_structure/overflow_file.cpp

```diff
@@ -0,0 +1,232 @@
+        lengthRead += numBytesToCheckInPage;
+    }
+    return true;
+}
+
+uint8_t* OverflowFileHandle::addANewPage() {
+    page_idx_t newPageIdx = overflowFile.getNewPageIdx();
+    if (pageCache.size() > 0) {
+        pageCache[nextPosToWriteTo.pageIdx]->write(
+            END_OF_PAGE, reinterpret_cast<uint8_t*>(&newPageIdx), sizeof(page_idx_t));
+    }
+    pageCache.emplace(newPageIdx, std::make_unique<InMemPage>());
+    nextPosToWriteTo.elemPosInPage = 0;
+    nextPosToWriteTo.pageIdx = newPageIdx;
+    return pageCache[newPageIdx]->data;
+}
+
+void OverflowFileHandle::setStringOverflow(
+    const char* srcRawString, uint64_t len, ku_string_t& diskDstString) {
+    if (len <= ku_string_t::SHORT_STR_LENGTH) {
+        return;
+    } else if (len > BufferPoolConstants::PAGE_256KB_SIZE) {
+        if constexpr (StorageConstants::TRUNCATE_OVER_LARGE_STRINGS) {
+            len = BufferPoolConstants::PAGE_256KB_SIZE;
+            diskDstString.len = len;
+        } else {
+            throw RuntimeException(ExceptionMessage::overLargeStringPKValueException(len));
+        }
+    }
+    uint8_t* pageToWrite = nullptr;
+    if (nextPosToWriteTo.pageIdx == INVALID_PAGE_IDX) {
+        pageToWrite = addANewPage();
+    } else {
+        auto cached = pageCache.find(nextPosToWriteTo.pageIdx);
+        if (cached != pageCache.end()) {
+            pageToWrite = cached->second->data;
+        } else {
+            overflowFile.readFromDisk(
+                TransactionType::WRITE, nextPosToWriteTo.pageIdx, [&](auto* frame) {
+                    auto page = std::make_unique<InMemPage>();
+                    memcpy(page->data, frame, BufferPoolConstants::PAGE_4KB_SIZE);
+                    pageToWrite = page->data;
+                    pageCache.emplace(nextPosToWriteTo.pageIdx, std::move(page));
+                });
+        }
+    }
+    int32_t remainingLength = len;
+    TypeUtils::encodeOverflowPtr(
+        diskDstString.overflowPtr, nextPosToWriteTo.pageIdx, nextPosToWriteTo.elemPosInPage);
+    while (remainingLength > 0) {
+        auto bytesWritten = len - remainingLength;
+        auto numBytesToWriteInPage = std::min(
+            static_cast<uint32_t>(remainingLength), END_OF_PAGE - nextPosToWriteTo.elemPosInPage);
+        memcpy(pageToWrite + nextPosToWriteTo.elemPosInPage, srcRawString + bytesWritten,
+            numBytesToWriteInPage);
+        remainingLength -= numBytesToWriteInPage;
+        nextPosToWriteTo.elemPosInPage += numBytesToWriteInPage;
+        if (nextPosToWriteTo.elemPosInPage >= END_OF_PAGE) {
+            pageToWrite = addANewPage();
+        }
+    }
+}
+
+ku_string_t OverflowFileHandle::writeString(std::string_view rawString) {
+    ku_string_t result;
+    result.len = rawString.length();
+    auto shortStrLen = ku_string_t::SHORT_STR_LENGTH;
+    auto inlineLen = std::min(shortStrLen, (uint64_t)result.len);
+    memcpy(result.prefix, rawString.data(), inlineLen);
+    overflowFile.setHeaderChanged();
```

Comment:
I don't follow L139. Why do we set header to changed here? Shouldn't we set it inside `setStringOverflow` when the string is not short?

Reply:
Yes